pkg/scalers/metrics_api_scaler.go

```diff
@@ -63,6 +124,61 @@ func metricsAPIMetadata(resolvedEnv, metadata, authParams map[string]string) (*m
 		return nil, fmt.Errorf("no valueLocation given in metadata")
 	}
 
+	// no authMode specified
+	if _, ok := authParams["authMode"]; !ok {
+		return &meta, nil
+	}
+
+	val, _ := authParams["authMode"]
```

Comment:
```suggestion
	val, ok := authParams["authMode"]
	if !ok {
		return &meta, nil
	}
```
How about consolidating this? Probably it would be also reasonable to rename `val` to something like `authParams`

Reply:
Especially that we are using `authParams["apiKey"]` multiple times in next lines

Reply:
makes sense. done!